src/resolution/resolver.ts

```diff
@@ -290,7 +290,7 @@ const _getParentContainersIterator = (container: interfaces.Container, includeSe
     return containersIterator;
 }
 
-function resolve<T>(context: interfaces.Context): T {
+function resolve<T>(context: interfaces.Context): (T | T[] | Promise<T> | Promise<T>[]) {
```

Comment:
The right return type is `(T | Promise<T> | (T | Promise<T>)[]))

Reply:
same thing.

Reply:
will change though

Reply:
Don't worry, I got it, can I push it? This way I can add more typings :)

The difference is subtle but important imo

Reply:
will do !

Reply:
I will push in a minute or two then is all yours ? Too tired ! Yes the difference is important, in particular to the your review below.

Reply:
All right, I'll add the changes now or tomorrow morning :) . I hope I'm not being rude. Sometimes it's hard for me to notice